### PR TITLE
Add Google Analytics and Privacy Policy

### PR DESCRIPTION
**Pull request by Accio1**
### Resolves:

Does not resolve a specific issue. This was discussed by @Accio1 @leahcimto and @MakeTheBrainHappy as a way to improve the site.

### Changes:

-Adds Google Analytics to track site interactions.
-Adds a privacy warning banner on the bottom of the page when you visit for the first time in a new month.
-Adds a privacy policy and a link to it in the footer.

### Local Tests:

Tested locally. Note: As far as I know, Google Analytics does not track local usage statistics, so I won't know if the code works until it is on the actual site.

The banner informing about Google Analytics should only show up once. Once you press "Okay" it will not show up again until next month. To test that this works, you can clear local data using the Chrome Inspect Element.

Reviewers: Also feel free to suggest wording changes to the Privacy Policy.

@leahcimto

### CODE REVIEW

**Review by leahcimto — APPROVED** [review]
Previewed it locally and it looks good!

Do you think there could be better line spacing in the privacy policy? (between each bullet)
Also, I'm not sure if this counts but you could add Google Analytics as a service on the contributors' page. 
Lastly, could the pop-up notice be put in the header so we don't have to put it on each page?

**Review by leahcimto — COMMENTED** [review]
One thing though, do you think there could be better line spacing in the privacy policy? (between each bullet)
Also, I'm not sure if this counts but you could add Google Analytics as a service on the contributors' page.

**Comment by leahcimto** [comment]
Also, I'm not sure if this counts but you could add Google Analytics as a service on the contributors' page.

File: privacy/index.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+				<li>Examples of data that Google Analytics may collect include: IP Address, Operating System, the amount of time that you spend on each page, and what you click while on the page.</li>
+				<li>This data is not shared or sold with anybody besides site developers, who use it to improve the site based on usage statistics.</li>
+				<li>If you do not agree to the use of Google Analytics, please discontinue your use of this site. Alternatively, consider installing the <a href="https://support.google.com/analytics/answer/181881?hl=en-GB">Google Analytics opt-out browser add-on</a>.</li>
+			</ul>
+
+			<h3>Other</h3>
+			<ul>
+				<li>Aside from Google Analytics, this site also stores two cookies on your computer which save whether or not you have agreed to the use of Google Analytics, and the date that you agreed.</li>
+				<li>This cookie expires each month, so you will have to re-accept the use of Google Analytics when you visit our site in a month that you have not yet accepted.</li>
+				<li>Questions? Feel free to contact us by opening a discussion on our GitHub repository.</li>
```
Comment (gosoccerboy5) [inline privacy/index.html:40]: Maybe add a link to the Github repo? I guess there's always a link in the footer, but..

Comment (gosoccerboy5) [reply privacy/index.html:40]: Other than that, looks great.

**Comment by Accio1** [comment]
> Previewed it locally and it looks good!
> 
> Do you think there could be better line spacing in the privacy policy? (between each bullet)

For sure! I forgot to change that.
> Also, I'm not sure if this counts but you could add Google Analytics as a service on the contributors' page.

Yep, I could
> Lastly, could the pop-up notice be put in the header so we don't have to put it on each page?

I'm not really sure what you mean by this. I put it as a separate "component" so that we can show or hide it based on whether the person has accepted or not.

File: privacy/index.html (hunk 1 above)
Comment (Accio1) [reply privacy/index.html:40]: Yep, I can add a link!

**Comment by leahcimto** [comment]
> I'm not really sure what you mean by this. I put it as a separate "component" so that we can show or hide it based on whether the person has accepted or not.

Ok

**Review by gosoccerboy5 — APPROVED** [review]
Looks great!

**Comment by Accio1** [comment]
@leahcimto @gosoccerboy5 I just made the changes that you requested.

**Review by leahcimto — APPROVED** [review]
Looks great! Just made a small commit that changed the link for opening a discussion to having it in the q&a category.

**Comment by Accio1** [comment]
> Looks great! Just made a small commit that changed the link for opening a discussion to having it in the q&a category.

Great! In the future I might make a category just for questions about privacy, but for now q&a is fine.

Merging!

**Comment by leahcimto** [comment]
Yay!

**Comment by MTM828** [comment]
But... why?

**Comment by Accio1** [comment]
> But... why?

@MTM828 in order to gain data about site usage. For example, which devices are being used to visit the site. Rest assured that your data is not sold or shared with anyone, except in the form of general statistics (eg. if you visit the site on a mobile device, you will be included in the statistic that __% of users visit the site on mobile). Also, your data is anonymized so there is no way for me to actually see any information like your IP Address or Location. Additionally, keep in mind that Scratch also uses Google Analytics.

**Comment by MTM828** [comment]
> @MTM828 in order to gain data about site usage. For example, which devices are being used to visit the site. Rest assured that your data is not sold or shared with anyone, except in the form of general statistics (eg. if you visit the site on a mobile device, you will be included in the statistic that __% of users visit the site on mobile). Also, your data is anonymized so there is no way for me to actually see any information like your IP Address or Location. Additionally, keep in mind that Scratch also uses Google Analytics.

OK, I understand.
But um... where can i find the stats?

**Comment by Accio1** [comment]
> > @MTM828 in order to gain data about site usage. For example, which devices are being used to visit the site. Rest assured that your data is not sold or shared with anyone, except in the form of general statistics (eg. if you visit the site on a mobile device, you will be included in the statistic that __% of users visit the site on mobile). Also, your data is anonymized so there is no way for me to actually see any information like your IP Address or Location. Additionally, keep in mind that Scratch also uses Google Analytics.
> 
> OK, I understand.
> But um... where can i find the stats?

You can't. Only I can view them because I set up the Google Analytics account.

**Comment by MTM828** [comment]
> You can't. Only I can view them because I set up the Google Analytics account.

Oh, OK LOL.

**Pull request merged**